docs/book/src/developer/providers/v1.1-to-v1.2.md

```diff
@@ -192,6 +192,9 @@ NOTES:
   - `WaitForControlPlaneAndMachinesReady`
   - `DiscoveryAndWaitForMachineDeployments`
 - The `AssertControlPlaneFailureDomains` function in the E2E test framework has been modified to allow proper failure domain testing.
+- The self-hosted upgrade test also upgrades the self-hosted clusters kubernetes version by default. For that it requires the variables
```

Comment:
I would propose to not cherry-pick this change and add this to v1.2-v1.3.

I think it's a bit late to cherry-pick something like this into v1.2